### DISCUSSION TITLE
Inverted ColorToHexConverter

### DISCUSSION THREAD

**Discussion by sissto** [AvaloniaUI/Avalonia#15586] [Ideas]
Hi all,
I'm using the `ColorPickerButton` which only accepts a value of type `Color`. But I need the hex code of the selected color. Therefore I was  happy to find the `HexToColorConverter`. Unfortunately it only converts in the wrong direction. So my idea is to add a new property for inverting the conversion. 

What do you think? May I miss something and there is already such a feature?

Best regards
sissto

**Comment by timunie** [comment]
should be there? https://github.com/AvaloniaUI/Avalonia/blob/ff021aa5d91e52f14d6a676aa9717680fa600770/src/Avalonia.Controls.ColorPicker/Converters/ColorToHexConverter.cs#L67-L75

**Reply by sissto** [reply]
Yes, it is in general but I cannot use it in a binding in the following scenario :(

In View:
`<ui:ColorPickerButton Color="{Binding HexColor, Converter={StaticResource HexConverter}}" />`

In ViewModel:
`[Reactive] public string HexColor { get; set; }`

**Reply by timunie** [reply]
How is `StaticResource HexConverter` defined?

**Reply by sissto** [reply]
Whoops, sorry.. It is defined like this:
```axaml
xmlns:converters="clr-namespace:Avalonia.Controls.Converters;assembly=Avalonia.Controls.ColorPicker"

<UserControl.Resources>
    <converters:ColorToHexConverter x:Key="HexConverter" /> 
</UserControl.Resources>
```

**Reply by timunie** [reply]
okay sometimes you need to do it to understand it. Yeah I believe a `HexToColorConverter` could be a useful addition. Do you want to file a PR for this?

/cc @robloo if you are against it?

**Reply by robloo** [reply]
I disagree with this so far. Are you really asking for an inverted `HexToColorConverter`?

Is your `HexColor` in the view model an `int` or a `string`? Generally, we should internally be using the most type-safe specific primitives to pass data internally. In this case you should be using `Color` or `HslColor` for the property on your view model and the binding will just work. This is easy to do on the view model level and at that layer you should already be using colors or brushes IMO.

Note that I kept `ColorToHexConverter` in the ColorPicker package because I didn't consider it general-purpose enough for wide use.

**Reply by sissto** [reply]
Yes I'm asking for an inverted `HexToColorConverter`.

The `HexColor`property is a `string` as you can see in the first reply of this post.

I totally understand your point of view. I get the hex code from a backend and I always try to keep my view models clean of any conversion code. To me a binding converter is the perfect solution for this, but no worries! Was just an idea and if you don't like it, it's ok for me :)

**Reply by robloo** [reply]
Yea, storing as a string internally is not ideal here. Not only is the overhead much higher than an int or Color it's a bit unsafe. This is why I think you should pursue another option.

That said, I am not fundamentally against a HexToColorConverter. It could make sense some place else.

**Comment by robloo** [comment]
Note the `ColorPickerButton` from the Windows Community Toolkit is just the `ColorPicker` here in Avalonia. The naming was aligned to match correctly with all other "pickers".

**Reply by sissto** [reply]
Yeah, to be fair the `ColorPickerButton` is from the FluentAvalonia package 😇
